debugger/src/debug_adapter.rs

```diff
@@ -991,15 +1018,15 @@ impl<P: ProtocolAdapter> DebugAdapter<P> {
                                         all_threads_stopped: Some(true),
                                         hit_breakpoint_ids: None,
                                     });
-                                    self.send_event("stopped", event_body);
+                                    self.send_event("stopped", event_body)?;
                                     new_status
                                 }
                                 other => other,
                             },
                             Err(_) => CoreStatus::Unknown,
                         };
                         self.last_known_status = core_status;
-                        true
+                        Ok(())
                     }
                 }
             }
```

Comment:
This is not related to this code change, but when the `Err(error)` is handled  on the next line, I think it should look something like this 
```
Err(error) => {
    self.last_known_status = CoreStatus::Halted(HaltReason::Unknown);
    Err(DebuggerError::Other(anyhow!("{}", error))) // Need to test that this error is caught by the caller, and that it results in a `DebuggerStatus::TerminateSession` being returned to Debugger::debug_session()
}
```
If the `Core::run()` fails, there really is no use in continuing with the current session. In such a case, I believe we should report the error and terminate the session.

Reply:
Ok, so it should look like this, probably:
```rust
Err(error) => {
    self.last_known_status = CoreStatus::Halted(HaltReason::Unknown);
    self.send_response::<()>(request, Err(DebuggerError::Other(anyhow!("{}", error))))?;
    Err(DebuggerError::Other(anyhow!("{}", error)))
}
```

Reply:
Yes (except the return is a Result<()>, so slight modification:
```
Err(error) => {
    self.last_known_status = CoreStatus::Halted(HaltReason::Unknown);
    self.send_response::<()>(request, Err(DebuggerError::Other(anyhow!("{}", error))))?;
    Err(DebuggerError::Other(anyhow!("{}", error)))
}
```

I tested this, and then if something causes the `Core::run()` to fail, it passes the error to the user, and ends the session, but keeps the debugger alive (when run in server mode). 

I will mark my review as `Approved`, so that you can merge once this code is updated.